### ISSUE TITLE
squash commits prevent branch chaining

### ISSUE THREAD

**Issue by seanmakesgames** [comcode-org/hackmud_wiki#80]
### Problem

Because of squashing PR commits, it is hard to create a branch off of an unmerged branch to continue work on it. When you go to merge the branch that was forked off of the original, you can't rebase easily (conflicts often)

### Context

This would be solved with rebase or merge commit methods for merging.

**Comment by seanmakesgames** [comment]
should discuss pros and cons here

**Comment by danswann** [comment]
Is there a discussion to be had about whether this:
> create a branch off of an unmerged branch to continue work on it

is something we actually want/expect to happen in our workflow?

That should help answer whether this is a problem to begin with.

**Comment by seanmakesgames** [comment]
yeah, that's the discussion, I think. Basically, the idea is, let's say you find a 'next issue that you might fix' but you want to follow our process around merging early and often, making sure that PRs are only solving one problem- and you need to build on your branch which is in an open PR to be able to continue developing this next fix -- that situation makes the merging much more complicated.

**Comment by danswann** [comment]
At first glance, it sounds like needing to create a new topic branch before the one it relies on is merged is a failure to merge early and/or often.

I want to unpack your problem a little more. Is it something like this?
- I have identified multiple problems I want to fix that depend on code in an open PR.
- I want to get to work on more of those problems right away, without waiting for the branch they depend on to get reviewed and  merged (i.e. because I don't want to lose productivity momentum or something).

To be clear: squash merge is not a hill I'm interested in dying on, but I think there's at least *some* merit to the idea that this has exposed issues with our process rather than being inherently wrong for our process.

**Comment by seanmakesgames** [comment]
100% agree. yeah, a bit of extrapolation:

- I have 3 hours available to work. I identify an issue that is high priority because it blocks a few other issues. 
I fix the first issue, create a PR, and no one is around to review it. I would love to continue working on the issues that are dependent on the first PR.
   - In this case, if the PR could be reviewed then we'd be good / fine.
- I started a PR, but during the review process we discovered that it actually solves multiple problems. We are mostly ok with solving those problems, but should break them out to other PRs / reviews. 
   - In this case, even if the PR is reviewed and merged quickly, breaking out the changes to another branch still has challenges
